src/NHibernate/Linq/ExpressionTransformers/LikeTransformer.cs

```diff
@@ -0,0 +1,60 @@
+using System.Linq.Expressions;
+using System.Reflection;
+using NHibernate.Util;
+using Remotion.Linq.Parsing.ExpressionVisitors.Transformation;
+
+namespace NHibernate.Linq.ExpressionTransformers
+{
+	/// <summary>
+	/// Replace <see cref="string.StartsWith(string)"/>, <see cref="string.EndsWith(string)"/> and <see cref="string.Contains(string)"/>
+	/// with <see cref="SqlMethods.Like(string, string)"/>
+	/// </summary>
+	internal class LikeTransformer : IExpressionTransformer<MethodCallExpression>
+	{
+		private static readonly MethodInfo Like = ReflectHelper.FastGetMethod(SqlMethods.Like, default(string), default(string));
+
+		public ExpressionType[] SupportedExpressionTypes { get; } = {ExpressionType.Call};
+
+		public Expression Transform(MethodCallExpression expression)
+		{
+			if (expression.Method == ReflectionCache.StringMethods.StartsWith)
+			{
+				return Expression.Call(
+					Like,
+					expression.Object,
+					Concat(expression.Arguments[0], Expression.Constant("%"))
```

Comment:
How escaping works for such queries? I mean do we really properly support queries like `Where(x => x.Name.StartsWith("%"))`?

Reply:
It is not supported at the moment.

Reply:
Related: #484, #488, [NH-3726](https://nhibernate.jira.com/browse/NH-3726), 9be7d064bddcf64d37e73e6716162ef43f2a72a2.
The `StartsWith`, `EndsWith`, `Contains` cases were a comment "side note" in NH-3726 and #484. The main point of [NH-3726](https://nhibernate.jira.com/browse/NH-3726) was only about `SqlMethod.Like`. #484 and #488 were PR doing this, and it seems the corresponding fix has ended up committed directly as 9be7d06.
Anyway we lack an opened issue for supporting escaping with these methods, and as said in #484, it will be a possible breaking change.

Reply:
No, we have @fredericDelaporte. #877